lib/sdf-server/src/service/v2/variant/generate_aws_asset_schema.rs

```diff
@@ -0,0 +1,94 @@
+use axum::extract::{Host, OriginalUri, Path, Query};
+use dal::{
+    schema::variant::authoring::VariantAuthoringClient, ChangeSet, ChangeSetId, SchemaVariant,
+    SchemaVariantId, WorkspacePk, WsEvent,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    extract::{AccessBuilder, AssetSprayer, HandlerContext, PosthogClient},
+    service::force_change_set_response::ForceChangeSetResponse,
+    track,
+};
+
+use super::SchemaVariantsAPIResult;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct AwsCommand {
+    pub command: String,
+    pub subcommand: String,
+}
+
+pub async fn generate_aws_asset_schema(
+    HandlerContext(builder): HandlerContext,
+    AssetSprayer(asset_sprayer): AssetSprayer,
+    AccessBuilder(access_builder): AccessBuilder,
+    PosthogClient(posthog_client): PosthogClient,
+    OriginalUri(original_uri): OriginalUri,
+    Host(host_name): Host,
+    Path((_workspace_pk, change_set_id, schema_variant_id)): Path<(
+        WorkspacePk,
+        ChangeSetId,
+        SchemaVariantId,
+    )>,
+    Query(aws_command): Query<AwsCommand>,
+) -> SchemaVariantsAPIResult<ForceChangeSetResponse<()>> {
+    let mut ctx = builder
+        .build(access_builder.build(change_set_id.into()))
+        .await?;
+
+    // Generate the code
+    let code = asset_sprayer
+        .aws_asset_schema(&aws_command.command, &aws_command.subcommand)
+        .await?;
+
+    // Update the function
+    let force_change_set_id = ChangeSet::force_new(&mut ctx).await?;
```

Comment:
This won't be an issue with your PR, but usually you see `force_change_set_id` at the top just in case `ctx` is used immediately. This isn't a blocker for me to move it higher... though as I'm saying this I'm realizing that footgun is big in many of our routes... not an issue for this PR though!

Reply:
Will do, I'm modifying other stuff here anyway.